fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateAliasNode.java

```diff
@@ -0,0 +1,79 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.analysis;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.plans.GroupPlan;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.logical.LogicalPlan;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Eliminate the logical sub query and alias node after analyze and before rewrite
+ * If we match the alias node and return its child node, in the execute() of the job
+ *
+ * TODO: refactor group merge strategy to support the feature above
+ */
+public class EliminateAliasNode implements AnalysisRuleFactory {
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+                RuleType.PROJECT_ELIMINATE_ALIAS_NODE.build(
+                        logicalProject().then(project -> eliminateSubQueryAliasNode(project, project.children()))
```

Comment:
why not use this pattern and save work to judge child type is LogicalSubqueryAlias?

```
logicalProject(logicalSubqueryAlias()).then(...)
```

Reply:
That is really ok！Thx.

Reply:
> why not use this pattern and save work to judge child type is LogicalSubqueryAlias?
> 
> ```java
> logicalProject(logicalSubqueryAlias()).then(...)
> ```
logicalJoin cannot use the pattern, others are ok

Reply:
i think logicalJoin could use
```
logicalJoin(logicalSubqueryAlias(), group())

+

logicalJoin(group(), logicalSubqueryAlias()) 
```

Reply:
or you can use
```java
logicalJoin(any(), any())
  .when(join -> join.left() instanceof LogicalSubqueryAlias || join.right() instanceof LogicalSubqueryAlias)
  .then(join -> ...)
```